### PR TITLE
Sprint 5/issue 46#editing show and edit reservations screen

### PR DESCRIPTION
**Pull request by ramondiaz21**
The view was created to be able to visualize individually each one of the reservations, as well as the view for the edition of each one of them.
The corresponding responsibility was added.
You can view the changes in the url http://127.0.0.1:3000/reservations/1
and at http://127.0.0.1:3000/reservations/1/edit
screencapture-127-0-0-1-3000-reservations-1-2024-04-12-12_41_50

screencapture-127-0-0-1-3000-reservations-1-edit-2024-04-12-12_41_33

### CODE REVIEW

File: app/assets/stylesheets/application.tailwind.css [hunk 1]
```diff
@@ -53,3 +53,4 @@
 .ts-dropdown .active {
   @apply bg-gray-100 text-gray-900;
 }
+@import 'actiontext.css';
```
Comment (MauricioNC) [inline app/assets/stylesheets/application.tailwind.css:56]: Add a new line at the end of your code

File: app/views/reservations/_form.html.erb [hunk 2]
```diff
@@ -1,51 +1,65 @@
...
+  <%= form.submit class: "reservations-buttons border-b-none flex items-center" do %>
+  <svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor"
+    class="w-6 h-6 mr-2">
+    <path stroke-linecap="round" stroke-linejoin="round"
+      d="M16.023 9.348h4.992v-.001M2.985 19.644v-4.992m0 0h4.992m-4.993 0 3.181 3.183a8.25 8.25 0 0 0 13.803-3.7M4.031 9.865a8.25 8.25 0 0 1 13.803-3.7l3.181 3.182m0-4.991v4.99" />
+  </svg>
+  Update reservation
+  <% end %>
+</div>
+<% end %>
```
Comment (MauricioNC) [inline app/views/reservations/_form.html.erb:65]: Add a new line at the end of your code

File: app/views/reservations/_reservation.html.erb [hunk 3]
```diff
@@ -1,24 +1,87 @@
 <%= content_tag :div, id: (dom_id reservation) do %>
-  <h1 class="font-bold text-4xl"><%= reservation.accommodation.title %></h1>
+<h2 class="text-3xl text-black mt-0">Your reservation: <%= reservation.accommodation.title %></h2>
+<div class="grid grid-cols-1 sm:grid-cols-2 lg:grid-cols-2 gap-4">
```
Comment (MauricioNC) [inline app/views/reservations/_reservation.html.erb:3]: Indent correctly this code

File: app/views/reservations/_form.html.erb [hunk 4]
```diff
@@ -1,51 +1,65 @@
 <%= form_with(model: reservation, class: "contents", 
                                   data: { controller: "reservationdatespicker", 
                                           reservationdatespicker_reservation_id_value: reservation.id }) do |form| %>
-  <% if reservation.errors.any? %>
-    <div id="error_explanation" class="bg-red-50 text-red-500 px-3 py-2 font-medium rounded-lg mt-3">
-      <h2><%= pluralize(reservation.errors.count, "error") %> prohibited this reservation from being saved:</h2>
+<% if reservation.errors.any? %>
+<div id="error_explanation" class="bg-red-50 text-red-500 px-3 py-2 font-medium rounded-lg mt-3">
+  <h2><%= pluralize(reservation.errors.count, "error") %> prohibited this reservation from being saved:</h2>
```
Comment (MauricioNC) [inline app/views/reservations/_form.html.erb:7]: Indent correctly this code

File: app/views/reservations/edit.html.erb [hunk 5]
```diff
@@ -1,8 +1,24 @@
...
+  <div class="buttons-wrapper">
+    <%= link_to(@reservation, class: "reservations-buttons border-b-none") do %>
+    <svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor"
+      class="w-6 h-6">
+      <path stroke-linecap="round" stroke-linejoin="round"
+        d="M2.036 12.322a1.012 1.012 0 0 1 0-.639C3.423 7.51 7.36 4.5 12 4.5c4.638 0 8.573 3.007 9.963 7.178.07.207.07.431 0 .639C20.577 16.49 16.64 19.5 12 19.5c-4.638 0-8.573-3.007-9.963-7.178Z" />
+      <path stroke-linecap="round" stroke-linejoin="round" d="M15 12a3 3 0 1 1-6 0 3 3 0 0 1 6 0Z" />
+    </svg>
+    Show this reservation
+    <% end %>
```
Comment (MauricioNC) [inline app/views/reservations/edit.html.erb:14]: Same, indent correctly this code

File: app/views/users/sessions/new.html.erb [hunk 6]
```diff
@@ -1,6 +1,6 @@
 <div class="contenedor log-in">
-  <div class="seccion left">
-    <%= image_tag 'Image_sidebar.jpg', class:"w-100 h-100"%>
+  <div class="seccion left"><%= image_tag 'Image_sidebar.jpg', class:"w-100 h-100"%>
+    
   </div>
```
Comment (MauricioNC) [inline app/views/users/sessions/new.html.erb:4]: Remove empty line or move image_tag to the next line

**Review by MauricioNC — APPROVED** [review]

File: app/assets/stylesheets/application.tailwind.css (hunk 1 above)
Comment (MauricioNC) [inline app/assets/stylesheets/application.tailwind.css:56]: Add a new line at the end of your code

File: app/views/reservations/_form.html.erb [hunk 7]
```diff
@@ -1,51 +1,65 @@
...
```
Comment (MauricioNC) [inline app/views/reservations/_form.html.erb:6]: Indent correctly this code

File: app/views/reservations/_form.html.erb (hunk 2 above)
Comment (MauricioNC) [inline app/views/reservations/_form.html.erb:65]: Add a new line at the end of your code

File: app/views/reservations/_reservation.html.erb (hunk 3 above)
Comment (MauricioNC) [inline app/views/reservations/_reservation.html.erb:3]: Indent correctly this code

File: app/views/reservations/_reservation.html.erb [hunk 8]
```diff
@@ -1,24 +1,87 @@
...
-    <hr class="mt-6">
-  <% end %>
+      <%= link_to reservations_path, class: "reservations-buttons" do %>
+      <svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor"
+        class="w-6 h-6">
+        <path stroke-linecap="round" stroke-linejoin="round" d="M15.75 19.5 8.25 12l7.5-7.5" />
+      </svg>
+      Back
+      <% end %>
+      <% end %>
```
Comment (MauricioNC) [inline app/views/reservations/_reservation.html.erb:80]: Indent correctly this code

File: app/views/reservations/edit.html.erb (hunk 5 above)
Comment (MauricioNC) [inline app/views/reservations/edit.html.erb:14]: Same, indent correctly this code

File: app/views/reservations/edit.html.erb [hunk 9]
```diff
@@ -1,8 +1,24 @@
...
+    <%= link_to reservations_path, class: "reservations-buttons" do %>
+    <svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor"
+      class="w-6 h-6">
+      <path stroke-linecap="round" stroke-linejoin="round" d="M15.75 19.5 8.25 12l7.5-7.5" />
+    </svg>
+    Back to reservations
+    <% end %>
```
Comment (MauricioNC) [inline app/views/reservations/edit.html.erb:21]: Same, indent correctly this code

File: app/views/users/sessions/new.html.erb (hunk 6 above)
Comment (MauricioNC) [inline app/views/users/sessions/new.html.erb:4]: Remove empty line or move image_tag to the next line

**Review by MauricioNC — CHANGES_REQUESTED** [review]

File: app/views/reservations/_reservation.html.erb [hunk 10]
```diff
@@ -1,24 +1,87 @@
...
+      Edit
+      <% end %>
+      <%= button_to reservation_path(@reservation), method: :delete, class: "reservations-buttons no-border" do %>
+      <svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="1.5" stroke="currentColor"
+        class="w-6 h-6">
+        <path stroke-linecap="round" stroke-linejoin="round"
+          d="m14.74 9-.346 9m-4.788 0L9.26 9m9.968-3.21c.342.052.682.107 1.022.166m-1.022-.165L18.16 19.673a2.25 2.25 0 0 1-2.244 2.077H8.084a2.25 2.25 0 0 1-2.244-2.077L4.772 5.79m14.456 0a48.108 48.108 0 0 0-3.478-.397m-12 .562c.34-.059.68-.114 1.022-.165m0 0a48.11 48.11 0 0 1 3.478-.397m7.5 0v-.916c0-1.18-.91-2.164-2.09-2.201a51.964 51.964 0 0 0-3.32 0c-1.18.037-2.09 1.022-2.09 2.201v.916m7.5 0a48.667 48.667 0 0 0-7.5 0" />
+      </svg>
+      Destroy
+      <% end %>
```
Comment (juperameza) [inline app/views/reservations/_reservation.html.erb:71]: same here with the indentatio, i reccommend looking for a formatting extension i think prettier works just have to format it

**Review by juperameza — CHANGES_REQUESTED** [review]

**Review by PudinDeveloper — APPROVED** [review]
Fix the issues already mentioned and also...

Here are some extensions I use that could be quite helpful for some of the issues mentioned by other mentors:

- [Prettier - Code formatter](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode)
- [Ruby Language Server](https://marketplace.visualstudio.com/items?itemName=Shopify.ruby-lsp)
- [Simple Ruby ERB](https://marketplace.visualstudio.com/items?itemName=vortizhe.simple-ruby-erb)
- [Trailing Spaces](https://marketplace.visualstudio.com/items?itemName=shardulm94.trailing-spaces)
- [Error Lens](https://marketplace.visualstudio.com/items?itemName=usernamehw.errorlens)
- [Path Intellisense](https://marketplace.visualstudio.com/items?itemName=christian-kohler.path-intellisense)

File: app/assets/stylesheets/application.tailwind.css (hunk 1 above)
Comment (DevLuis123) [reply app/assets/stylesheets/application.tailwind.css:56]: solved!

Comment (DevLuis123) [reply app/assets/stylesheets/application.tailwind.css:56]: done!

File: app/views/reservations/_reservation.html.erb (hunk 8 above)
Comment (DevLuis123) [reply app/views/reservations/_reservation.html.erb:80]: done!

File: app/views/reservations/_form.html.erb (hunk 7 above)
Comment (DevLuis123) [reply app/views/reservations/_form.html.erb:6]: done!

File: app/views/reservations/edit.html.erb (hunk 5 above)
Comment (DevLuis123) [reply app/views/reservations/edit.html.erb:14]: done!

**Review by MauricioNC — APPROVED** [review]

**Review by juperameza — APPROVED** [review]

**Pull request merged**